platform/darwin/src/MGLImageSource.h

```diff
@@ -0,0 +1,59 @@
+#import "MGLSource.h"
+
+#import "MGLFoundation.h"
+#import "MGLTypes.h"
+#import "MGLGeometry.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+MGL_EXPORT
+@interface MGLImageSource : MGLSource
+
+#pragma mark Initializing a Source
+
+/**
+ Returns an image source with an identifier and coordinates
+
+ @param identifier A string that uniquely identifies the source.
+ @param coordinates the NW, NE, SW, and SE coordiantes for the image.
+ @return An initialized image source.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier coordinates:(MGLCoordinateQuad)coordinates NS_DESIGNATED_INITIALIZER;
+
+/**
+ Returns an image source with an identifier, coordinates and a URL
+ 
+ @param identifier A string that uniquely identifies the source.
+ @param coordinates the NW, NE, SW, and SE coordiantes for the image.
+ @param url An HTTP(S) URL, absolute file URL, or local file URL relative to the
+    current application’s resource bundle.
+ @return An initialized shape source.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier coordinates:(MGLCoordinateQuad)coordinates URL:(NSURL *)url;
+
+/**
+ Returns an image source with an identifier, coordinates and an image
+ 
+ @param identifier A string that uniquely identifies the source.
+ @param coordinates the NW, NE, SW, and SE coordiantes for the image.
+ @param image The image to display for the sourcde.
+ @return An initialized shape source.
+ */
+- (instancetype)initWithIdentifier:(NSString *)identifier coordinates:(MGLCoordinateQuad)coordinates image:(MGLImage *)image;
+
+#pragma mark Accessing a Source’s Content
+
+/**
+ The URL to the source image.
+
+ If the receiver was initialized using `-initWithIdentifier:coordinates:` or 
+ `-initWithIdentifier:coordinates:Image:`, this property is set to `nil`.
+ */
+@property (nonatomic, copy, nullable)NSURL *URL;
+
+@property (nonatomic, nullable, setter=setImage:)MGLImage *image;
```

Comment:
The only reason for maintaining parallel state would be as a last resort for implementing an image getter. `mbgl::style::ImageSource::getImage()` would make that unnecessary.

Having an image getter would make MGLImageSource consistent with `-[MGLStyle imageForName:]` for non-georeferenced images. It would also enable some of the same use cases as that method. Off the top of my head, an application could get the image from the source, manipulate it, and pass the manipulated image back to the source. Or the user could tap on an image source and the application would display the “selected” source’s image in an image view on the side. The application doesn’t have to keep track of the image itself; mbgl remains the source of truth.

Continuously modifying the source data could in some cases be better served by video sources; on the other hand, it’s much more feasible for an application to programmatically generate an image than a video.

Are there technical reasons against adding a `getImage()` in mbgl?